secrets/secrets.go

```diff
@@ -298,3 +324,45 @@ func NewSecrets(r io.Reader) (*Secrets, error) {
 	}
 	return secrets, nil
 }
+
+func csiPathParser(path string, inputSecrets Document) (secretsDocument Document, err error) {
+	secretsDocument = Document{
+		Secrets: make(map[string]GenericSecret),
+	}
+	file, err := os.Open(path)
+	if err != nil {
+		return secretsDocument, err
+	}
+	fileInfo, err := file.Stat()
+	if err != nil {
+		return secretsDocument, err
+	}
+	if fileInfo.IsDir() {
```

Comment:
you have a helper for this right?

Reply:
that helper function is in another package and is unexported :) and we likely don't want to export it (e.g. having `IsDirectory` being part of `filewatcher` api doesn't really make sense)

but again, since `NewSecrets` takes an `io.Reader` as an arg, I'm not really sure how we could get the error to say it's actually a directory not a file (you can't get an `io.Reader` out of a directory can you?) and get into here.

Reply:
This specific dir check is used to determine which whether the files within the parent dir are files during the recursion. And like Yuxuan said, that helper I made is in the other package